compiler/plugins/input/Torch/PluginRegistration.cpp

```diff
@@ -54,7 +58,11 @@ struct TorchSession
     registry.insert<torch::TorchConversion::TorchConversionDialect>();
     registry.insert<mlir::torch::TMTensor::TMTensorDialect>();
     registry.insert<mlir::ml_program::MLProgramDialect>();
+    registry.insert<IREE::Flow::FlowDialect>();
+    registry.insert<IREE::HAL::HALDialect>();
     registry.insert<IREE::LinalgExt::IREELinalgExtDialect>();
+    registry.insert<IREE::Stream::StreamDialect>();
+    registry.insert<IREE::Util::UtilDialect>();
```

Comment:
Does the torch plugin itself use Flow/Stream/HAL? The StableHLO and TOSA plugins only register what is unique to them: https://github.com/iree-org/iree/blob/deb8435af7079f71b782e21613fd4196203e052b/compiler/plugins/input/StableHLO/PluginRegistration.cpp#L66-L71 https://github.com/iree-org/iree/blob/deb8435af7079f71b782e21613fd4196203e052b/compiler/plugins/input/TOSA/PluginRegistration.cpp#L36-L38

I'd like to keep these registrations minimal... though the cost of adding extra dialects isn't too high.

Reply:
note that iree_input was the way we avoided this, but it doesn't make sense when in-tree - it still would pull in all the same deps, just via iree_input's lowerings. out-of-tree is different but we don't so much care about that right now enough to have a whole layer, and there are probably better solutions than when the external projects dialects were created.

Reply:
Gotcha. Thanks for explaining.

Reply:
I may still not be right - I just spent a whole day struggling with it and think this is what I discovered :P Always open to improvements here and I agree with you that the deps aren't quite in the right place!

Reply:
added a TODO